src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPLIT_REGEXP = "\\W+";
+    private static final String WORD_TO_DETECT_SUPPLY_COUNTER_NUMBER = "supply";
+    private static final String WORD_TO_DETECT_BUY_COUNTER_NUMBER = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
```

Comment:
this method should contain three calls of other methods: readFile() - returns content, generateReport() -(you pass contents here), writeToFile()-(you pass report and filename here ), this will be a good code decomposition

Reply:
Just to clarify, its for Single Responsibility, isn`t it?